core/client/app/controllers/setup/three.js

```diff
@@ -2,40 +2,130 @@ import Ember from 'ember';
             return validator.isEmail(user);
         });
     }),
-    numUsers: Ember.computed('usersArray', function () {
-        return this.get('usersArray').length;
+    validateUsers: Ember.computed('usersArray', function () {
+        var errors = [];
+
+        this.get('usersArray').forEach(function (user) {
+            if (!validator.isEmail(user)) {
+                errors.push({
+                    user: user,
+                    error: 'email'
+                });
+            }
+        });
+
+        return errors.length === 0 ? true : errors;
+    }),
+    numUsers: Ember.computed('validUsersArray', function () {
+        return this.get('validUsersArray').length;
+    }),
+    buttonText: Ember.computed('usersArray', function () {
+        var num = this.get('usersArray').length,
+            user;
+
+        if (num > 0) {
+            user = num === 1 ? 'user' : 'users';
+            user = num + ' ' + user;
+        } else {
+            user = 'some users';
+        }
+
+        return 'Invite ' + user;
     }),
-    buttonText: Ember.computed('numUsers', function () {
-        var user = this.get('numUsers') === 1 ? 'user' : 'users';
-        return this.get('numUsers') > 0 ?
-            'Invite ' + this.get('numUsers') + ' ' + user : 'I\'ll do this later, take me to my blog!';
+    buttonClass: Ember.computed('validateUsers', 'numUsers', function () {
+        if (this.get('validateUsers') === true && this.get('numUsers') > 0) {
+            return 'btn-green';
+        } else {
+            return 'btn-minor';
+        }
     }),
-    buttonClass: Ember.computed('numUsers', function () {
-        return this.get('numUsers') > 0 ? 'btn-green' : 'btn-minor';
+    authorRole: Ember.computed(function () {
+        return this.store.find('role').then(function (roles) {
+            return roles.findBy('name', 'Author');
+        });
     }),
     actions: {
         invite: function () {
-            console.log('inviting', this.get('usersArray'));
+            var self = this,
+                validationErrors = this.get('validateUsers'),
+                users = this.get('usersArray'),
+                errorMessages,
+                notifications = this.get('notifications');
+            console.log('NOTIFICATIONS', notifications);
 
-            if (this.get('numUsers') === 0) {
-                this.sendAction('signin');
-            }
+            if (validationErrors === true && users.length > 0) {
+                this.get('authorRole').then(function (authorRole) {
+                    Ember.RSVP.Promise.all(
+                        users.map(function (user) {
+                            var newUser = self.store.createRecord('user', {
+                                email: user,
+                                status: 'invited',
+                                role: authorRole
+                            });
+
+                            return newUser.save().then(function () {
```

Comment:
There should probably be some error handling here.  As it stands if this save fails there's no feedback shown on the page at all.  It just appears as though nothing has happened.  You can test it by trying to invite the email address you just signed up with.